app/src/pages/settings/GenerativeProvidersCard.tsx

```diff
@@ -0,0 +1,256 @@
+
+import {
+  Button,
+  Flex,
+  Icon,
+  Icons,
+  Input,
+  Label,
+  Text,
+  TextField,
+  View,
+} from "@phoenix/components";
+import {
+  GenerativeProviderIcon,
+  GenerativeProviderIconProps,
+} from "@phoenix/components/generative";
+import { tableCSS } from "@phoenix/components/table/styles";
+import { useCredentialsContext } from "@phoenix/contexts/CredentialsContext";
+import { assertUnreachable } from "@phoenix/typeUtils";
+
+import {
+  GenerativeProvidersCard_data$data,
+  GenerativeProvidersCard_data$key,
+} from "./__generated__/GenerativeProvidersCard_data.graphql";
+
+type ProviderKey =
+  GenerativeProvidersCard_data$data["modelProviders"][number]["key"];
+
+function ProviderIcon({ providerKey }: { providerKey: ProviderKey }) {
+  let provider: GenerativeProviderIconProps["provider"];
+  switch (providerKey) {
+    case "OPENAI":
+      provider = "OPENAI";
+      break;
+    case "ANTHROPIC":
+      provider = "ANTHROPIC";
+      break;
+    case "GEMINI":
+      provider = "GEMINI";
+      break;
+    case "AZURE_OPENAI":
+      provider = "AZURE_OPENAI";
+      break;
+    default:
+      assertUnreachable(providerKey);
+  }
+  return <GenerativeProviderIcon provider={provider} height={18} />;
+}
+
+export function GenerativeProvidersCard({
+  query,
+}: {
+  query: GenerativeProvidersCard_data$key;
+}) {
+  const [selectedProvider, setSelectedProvider] = useState<
+    GenerativeProvidersCard_data$data["modelProviders"][number] | null
+  >(null);
+  const credentials = useCredentialsContext((state) => state);
+  const setCredential = useCredentialsContext((state) => state.setCredential);
+  const data = useFragment<GenerativeProvidersCard_data$key>(
+    graphql`
+      fragment GenerativeProvidersCard_data on Query {
+        modelProviders {
+          name
+          key
+          dependenciesInstalled
+          dependencies
+          apiKeyEnvVar
+          apiKeySet
+        }
+      }
+    `,
+    query
+  );
+
+  const tableData = useMemo(() => [...data.modelProviders], [data]);
+  type DataRow = (typeof tableData)[number];
+  const columns = useMemo(() => {
+    return [
+      {
+        header: "Name",
+        accessorKey: "name",
+        cell: ({ row }) => {
+          return (
+            <Flex direction="row" alignItems="center" gap="size-100">
+              <ProviderIcon providerKey={row.original.key} />
+              {row.original.name}
+            </Flex>
+          );
+        },
+      },
+      {
+        header: "Environment Variable",
+        accessorKey: "apiKeyEnvVar",
+        cell: ({ row }) => {
+          return <Text>{row.original.apiKeyEnvVar}</Text>;
+        },
+      },
+      {
+        header: "configuration",
+        accessorKey: "apiKeySet",
+        cell: ({ row }) => {
+          if (!row.original.dependenciesInstalled) {
+            return <Text color="warning">missing dependencies</Text>;
+          }
+          if (row.original.apiKeySet) {
```

Comment:
What happens if the server has it configured and local has it configured? We just show that server has it configured?

Reply:
Can confirm, this is what it does. Slightly confusing when I see "configured on the server", then I edit it locally on the same screen, and it never updates to say its local

Reply:
ah, yeah I'll show set locally